source/OctoPack.Tasks/Util/AssemblyExtensions.cs

```diff
@@ -16,8 +26,19 @@ public static string FullLocalPath(this Assembly assembly)
         return root;
     }
 
+
     public static string GetFileVersion(this Assembly assembly)
     {
         return assembly.GetCustomAttributes(true).OfType<AssemblyFileVersionAttribute>().First().Version;
     }
-}
+
+
+    public static string GetNugetVersion(this Assembly assembly)
+    {
+        var gitVersionInformationType = assembly.GetType("GitVersionInformation");
+        if (gitVersionInformationType == null)
+            return null;
+        var versionField = gitVersionInformationType.GetField("NuGetVersion");
+        return versionField.GetValue(null) as string;
```

Comment:
Implemented suggestion. Also redid the history to one single commit containing the whole PR